driver-sync/src/test/functional/com/mongodb/client/AbstractClientSideEncryptionNotSpawnMongocryptdTest.java

```diff
@@ -0,0 +1,244 @@
+    @Test
+    void viaBypassAutoEncryption() {
+        assumeFalse(CRYPT_SHARED_LIB_PATH_SYS_PROP_VALUE != null);
+        setUpCollection((mongocryptdSocketAddress, autoEncryptionSettingsBuilder) -> autoEncryptionSettingsBuilder
+                .extraOptions(commonExtraAutoEncryptionOptions(mongocryptdSocketAddress))
+                .bypassAutoEncryption(true)
+        ).insertOne(Document.parse("{unencrypted: 'test'}"));
+        assertMongocryptdNotSpawned();
+    }
+
+    /**
+     * See
+     * <a href="https://github.com/mongodb/specifications/tree/master/source/client-side-encryption/tests#via-bypassqueryanalysis">
+     * Via bypassQueryAnalysis</a>.
+     */
+    @Test
+    void viaBypassQueryAnalysis() {
+        assumeFalse(CRYPT_SHARED_LIB_PATH_SYS_PROP_VALUE != null);
+        setUpCollection((mongocryptdSocketAddress, autoEncryptionSettingsBuilder) -> autoEncryptionSettingsBuilder
+                .extraOptions(commonExtraAutoEncryptionOptions(mongocryptdSocketAddress))
+                .bypassQueryAnalysis(true)
+        ).insertOne(Document.parse("{unencrypted: 'test'}"));
+        assertMongocryptdNotSpawned();
+    }
+
+    private MongoCollection<Document> setUpCollection(
+            final BiConsumer<InetSocketAddress, AutoEncryptionSettings.Builder> autoEncryptionSettingsBuilderMutator) {
+        setUpKeyVaultNamespace();
+        InetSocketAddress localMongocryptdSocketAddress = new InetSocketAddress(
+                InetAddress.getLoopbackAddress(), findAvailableMongocryptdLoopbackPort());
+        AutoEncryptionSettings.Builder autoEncryptionSettingsBuilder = AutoEncryptionSettings.builder();
+        autoEncryptionSettingsBuilderMutator.accept(localMongocryptdSocketAddress, autoEncryptionSettingsBuilder);
+        mongocryptdSocketAddress = localMongocryptdSocketAddress;
+        client = createMongoClient(MongoClientSettings.builder(getMongoClientSettings())
+                .autoEncryptionSettings(autoEncryptionSettingsBuilder
+                        .kmsProviders(singletonMap(LOCAL_KMS_PROVIDER_ID, singletonMap("key", localKmsProviderKey())))
+                        .keyVaultNamespace(KEY_VAULT_NAMESPACE.getFullName())
+                        .schemaMap(singletonMap(NAMESPACE.getFullName(), EXTERNAL_SCHEMA))
+                        .build())
+                .build());
+        MongoDatabase db = client.getDatabase(NAMESPACE.getDatabaseName());
+        db.drop();
+        return db.getCollection(NAMESPACE.getCollectionName());
+    }
+
+    private void setUpKeyVaultNamespace() {
+        try (MongoClient client = createMongoClient(MongoClientSettings.builder().build())) {
+            MongoDatabase db = client.getDatabase(KEY_VAULT_NAMESPACE.getDatabaseName());
+            db.drop();
+            db.getCollection(KEY_VAULT_NAMESPACE.getCollectionName(), BsonDocument.class)
+                    .withWriteConcern(WriteConcern.MAJORITY)
+                    .insertOne(EXTERNAL_KEY);
+        }
+    }
```

Comment:
The prose tests do not require having any keys in the `KEY_VAULT_NAMESPACE`: only `viaMongocryptdBypassSpawn` attempts to insert an encrypted field, and that insertion is supposed to fail. However, when the driver misbehaves or tests fail (in my case, `viaMongocryptdBypassSpawn` was failing if `viaLoadingSharedLibrary` runs before it in the same process due to `viaLoadingSharedLibrary` loading `crypt_shared`), things may get too difficult to understand if the encryption is not properly set up.